### ISSUE TITLE
`is_conjugate` for matrix groups

### ISSUE THREAD

**Issue by joschmitt** [oscar-system/Oscar.jl#1974]
**Describe the bug**
There is a special implementation for `is_conjugate(::Group, ::Elem, ::Elem)` for matrix groups in https://github.com/oscar-system/Oscar.jl/blob/7eeb2f39789c7ce9962d3885629840fb6e656caa/src/Groups/matrices/linear_isconjugate.jl#L145 which unnecessarily throws an error if the group is not GL_n or SL_n.

**To Reproduce**
```
julia> M = matrix(QQ, 1, 1, [ -1 ])
[-1]

julia> G = matrix_group(M)
Matrix group of degree 1 over Rational Field

julia> is_conjugate(G, G(M), G(M))
ERROR: ArgumentError: Group must be general or special linear group
Stacktrace:
 [1] is_conjugate(G::MatrixGroup{fmpq, fmpq_mat}, x::MatrixGroupElem{fmpq, fmpq_mat}, y::MatrixGroupElem{fmpq, fmpq_mat})
   @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/linear_isconjugate.jl:146
 [2] top-level scope
   @ REPL[4]:1

```

**Expected behavior**
If one comments out `is_conjugate(::MatrixGroup,...)` it works with the generic `is_conjugate`.

**System (please complete the following information):**
```
julia> Oscar.versioninfo(full=true)
OSCAR version 0.11.4-DEV - #master, 35ac2587be -- 2023-02-21 09:33:31 +0100
  combining:
    AbstractAlgebra.jl   v0.27.10
    GAP.jl               v0.9.3
    Hecke.jl             v0.17.5
    Nemo.jl              v0.32.7
    Polymake.jl          v0.9.0
    Singular.jl          v0.16.1
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.4+0
    GAP_jll                 v400.1200.200+0
    Singular_jll            v403.101.500+0
    libpolymake_julia_jll   v0.9.0+0
    libsingular_julia_jll   v0.29.0+0
    polymake_jll            v400.900.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 12 × Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
  Threads: 1 on 12 virtual cores
Environment:
  JULIA_EDITOR = vim
Official https://julialang.org/ release

```

**Comment by ThomasBreuer** [comment]
Several problems occur in this example (as usual).

The first is that the `is_conjugate` method for groups that know to be a GL or SL over some finite field seems to assume a GAP like method dispatch (which allows for "TryNextMethod").
Currently I have no better idea than to move the check whether the group knows that it is a GL or SL to the generic `is_conjugate` method.

The second problem is that the return value of the special `is_conjugate` method is not just a Boolean but a pair.
(Some tests for subgroup conjugacy also expect this format, which was not observed because Julia returns `true` when one asks for `true[1]`.)

I will provide a pull request for a (preliminary) fix.

**Comment by fingolfin** [comment]
Fixed by #2006
